.github/workflows/release.yaml

```diff
@@ -0,0 +1,338 @@
+name: release
+
+# TODO - the real trigger
+# on:
+#   push:
+#     tags:
+#       - 'v*'

```

Comment:
Should we build on release instead? I believe that is possible. We should check what permissions are set for tag push in this case.

Reply:
I don't believe it's possible to trigger a build on a draft release.  The flow we're thinking is:

1. [Optional] Create a draft release specifying the intended tag, author the draft release notes, etc.
2. Git tag and push the tag 
3. CI triggers at this point, and builds all the artifacts, then uploads them idempotently to the pre-existing release (if you did step 1)
4. Smoke/spot test the artifacts to verify things are good (eventually we'll automate this)
5. Update to Pre-release (drop draft status)
6. Broader testing
7. Mark release latest / drop pre-release.

Reply:
imo draft prs should not trigger ci, that should be the role of a prerelease

Reply:
> imo draft prs should not trigger ci

Can you clarify?  

This PR is wired up to trigger CI in PR mode just to test the plumbing, but that's definitely not how it will work once we're ready.  The "draft" I was mentioning above is a draft release which makes it invisible to non-maintainers.

Reply:
Sorry my brain and fingers are in different places. I meant draft releases

Reply:
Got it, yes, to clarify step 2 is not automatic.  A human has to do that.